src/scene/materials/shader-graph-node.js

```diff
@@ -0,0 +1,615 @@
+import { Texture } from '../../graphics/texture.js';
+import { Vec2 } from '../../math/vec2.js';
+import { Vec3 } from '../../math/vec3.js';
+import { Vec4 } from '../../math/vec4.js';
+

```

Comment:
overall, I would like to see more comments in this file.

Reply:
I can do a pass and add comments to anything that seems non-obvious

Reply:
more comments added